File: examples/go/covenant-vault/CovenantVault_test.go
package contract

import (
	"testing"
	runar "github.com/icellan/runar/packages/runar-go"
)

func newVault() *CovenantVault {
	return &CovenantVault{
		Owner:     runar.MockPubKey(),
		Recipient: runar.Hash160(runar.MockPubKey()),
		MinAmount: 1000,
	}
}

// Native execution tests for CovenantVault are limited because the
// covenant rule (hash256(output) == extractOutputHash(txPreimage))
// requires a real sighash preimage with matching hashOutputs. The
// mock preimage doesn't produce a meaningful hashOutputs, so we
// only test compilation here. The contract logic is fully verified
// by the TS test suite and conformance golden files.

func TestCovenantVault_Compile(t *testing.T) {
	if err := runar.CompileCheck("CovenantVault.runar.go"); err != nil {
		t.Fatalf("Rúnar compile check failed: %v", err)
	}
}

File: examples/rust/covenant-vault/CovenantVault_test.rs
// Native module import is omitted because the contract uses hex string
// literals (e.g., "1976a914") that represent ByteString values in the
// Rúnar DSL but are &str in native Rust. The contract logic is verified
// by the TS test suite and conformance golden files.

#[test]
fn test_compile() {
    runar::compile_check(
        include_str!("CovenantVault.runar.rs"),
        "CovenantVault.runar.rs",
    ).unwrap();
}
